### DISCUSSION TITLE
Infer allDay:: true if there is no start/end time on an inline event

### DISCUSSION THREAD

**Discussion by dxcore35** [davish/obsidian-full-calendar#374] [Feature Requests]
# ALL DAY EVENT

- Please remove the `[allDay:: true]`
- Make all day event by default if there is task with no additional time

# TIMED EVENT

- Please change formatting and logic of [startTime:: XX:XX]  [endTime:: XX:XX]
- Use standard task format ` HH-mm` and variable defining length

- [ ] Cleaning of code 12:05 duration: 1.5h

You should please utilize the current plugins naming convention, especialy from the https://obsidian-tasks-group.github.io/obsidian-tasks/ and also make the writing of the task as less burden as possible. The colaboration with Obsidian task developer will be very fruitfull, as he already created GUI for editing task properties. He is just missing the lenght parameter!

Thank you for again and again amazing improvement of the plugin!

**Comment by davish** [comment]
@dxcore35, thank you for submitting, but please submit one feature request per issue.

**Comment by davish** [comment]
I have changed the title to reflect the first requested issue. I won't be changing the time format from startTime/endTime at this point. While startTime/duration would also work, the goal is for inline event metadata to be as close to the metadata for frontmatter events as possible.

**Comment by pm64** [comment]
@davish thanks for everything you do. I'd like to suggest another goal for inline event syntax: to eliminate or reduce friction for new users that already use their daily note to schedule tasks, a very common use case. If you agree that minimizing this friction should also be a priority, and seek to find an approach that balances frontmatter event syntax consistency with this priority, then OP's request to "Make all day event by default if there is task with no additional time" could make sense.

Prior to installing this plugin, it would not occur to a user to include `[allDay:: true]` at the end of every daily note task. By making this the default, daily note events would automatically appear on the calendar for these users without any additional work.

But aside from this, it's very common for a task list to consist entirely of "all day" tasks. In these cases, the "allDay true" indicator on each task can create substantial visual noise, and making "allDay true" the default gives the user a way to eliminate it. The fact that the item is a task appearing under the user-specified daily note header makes it clear that the user wants the item displayed on their calendar, regardless of whether the timing is specified.

**Comment by HerrRau** [comment]
I'd like to second the suggestion "Make all day event by default if there is task with no additional time": The .ics calendar sources I'm using have entries with dates (that show up perfectly fine in Full Calendar) and entries without dates, to be interpreted as all day events, but apparently lacking some kind of all day setting information, so they simply don't show up in the Calendar.
